### PR TITLE
Initial build

### PR DESCRIPTION
**Pull request by adam-thomas**
@incuna/backend So setting up `incuna-pagination` was a bit of a pain.  I had to copy and paste from several different places, then modify, then fiddle, etc.  That sucks, so here's a repository of everything one might need to start up a new library.

As far as I can tell, it should be straightforwardly plug-and-play, with the only changes you absolutely need to make highlighted both in the readme and with `TODO`s that can be searched for.  The `test_project` setup works remarkably easily, which should mean no more faffing around writing test runners or installing the library into other projects to build migrations.  It also lets you run the project as a Django server by itself to check admin behaviour or view templates in isolation.

The full description's [in the readme](https://github.com/incuna/base-library/blob/in-the-beginning/README.md) if you want to know how it's meant to work.

Please review/merge when you're back at work :)

Just for clarity, this repo isn't meant to be released to PyPI or activated on Travis, but it does contain all the necessary basic config for that.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+language: python
+python:
+  - "2.7"
+  - "3.3"
+  - "3.4"
```
Comment (LilyFirefly) [inline .travis.yml:4]: +`3.5` -`3.3`

File: test_project/settings.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+DEBUG = True
+ALLOWED_HOSTS = []
+ROOT_URLCONF = 'TODO_PACKAGE_NAME.tests.urls'
+STATIC_URL = '/static/'
+
+SECRET_KEY = 'u!%l9f3x5yjza&u-r1cxsg=yhmq7)ng(9-nz-$r!v)ml8$htpu'
```
Comment (LilyFirefly) [inline test_project/settings.py]: Better to use something explicitly not for production. eg `'not-for-production'`

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+setup(
+    version='0.1.0',
+    name='TODO_PROJECT_NAME',
+    packages=find_packages(),
+    include_package_data=True,
+    install_requires=[],
+    description='TODO',
+    author='Incuna Ltd',
+    author_email='admin@incuna.com',
+    url='https://github.com/incuna/TODO_PROJECT_NAME',
```
Comment (LilyFirefly) [inline setup.py:13]:
``` python
    classifiers=[TODO_TROVE_CLASSIFIERS],
```

**Comment by adam-thomas** [comment]
Updated!

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    classifiers=[
+        'Development Status :: 5 - Production/Stable',
```
Comment (LilyFirefly) [inline setup.py]: I would probably have a different default for this - a completely new library usually won't start stable. It's a bit different for something pulled out of a project though, so YMMV.

Comment (adam-thomas) [inline setup.py]: I don't know what the other options are but if you have a more suitable suggestion I'd be happy to go with it.

Comment (LilyFirefly) [inline setup.py]: Probably [4 - Beta](https://pypi.python.org/pypi?:action=list_classifiers).

Comment (LilyFirefly) [inline setup.py]: You could list multiple and then people could delete as appropriate.

Comment (adam-thomas) [inline setup.py]: ^ That's a good idea.

Comment (adam-thomas) [inline setup.py]: I put a `TODO` in there and listed them in the readme.

Comment (LilyFirefly) [inline setup.py]: :+1: Might be a good idea to link to the full list of classifiers too.

Comment (adam-thomas) [inline setup.py]: :+1:

Comment (adam-thomas) [inline setup.py]: Wow, that is a _very_ small list of languages!

Comment (adam-thomas) [inline setup.py]:
```
Topic :: Games/Entertainment :: Fortune Cookies
Topic :: Games/Entertainment :: Multi-User Dungeons (MUD)
```

Hey Trove!  The past called, they said they had a Geocities page for you... :P

**Comment by adam-thomas** [comment]
Moar Trove added!

**Comment by adam-thomas** [comment]
Ta :)

File: setup.py (hunk 4 above)
Comment (meshy) [inline setup.py]: fyi https://neocities.org/

Comment (adam-thomas) [inline setup.py]: WAHT

Comment (adam-thomas) [inline setup.py]: I must host gifs.

**Pull request merged**